### PR TITLE
use docker system info for client metadata if possible

### PR DESCRIPTION
**Pull request by thrau**
This PR uses the docker socket (if available) to determine a machine id and the system. This gives us information about the host rather than only about the container. The machine ids are now also prefixed so we can determine how they were generated. Note that we always hash any real machine IDs for privacy purposes.

* `gen_` generated by localstack
* `dkr_` docker daemon id
* `sys_` machine id from the system file `/etc/machine-id`

Moreover, the PR uses the `LOCALSTACK_AUTH_TOKEN` as `api_key` field if available in preparation for our new licensing scheme.

### CODE REVIEW

File: localstack/utils/analytics/metadata.py [hunk 1]
```diff
@@ -101,32 +102,58 @@ def _generate_session_id() -> str:
 
 
 def _generate_machine_id() -> str:
+    try:
+        # try to get a robust ID from the docker socket
+        from localstack.utils.docker_utils import DOCKER_CLIENT
+
+        docker_id = DOCKER_CLIENT.get_system_id()
```
Comment (dfangl) [inline localstack/utils/analytics/metadata.py:109]: We could do something like `if DOCKER_CLIENT.get_system_id() == DOCKER_CLIENT.get_system_id():` here, to check if we are in Docker (ID reliable) or in podman (generating an new one every time, then we could as well use one of the fallbacks).

File: localstack/utils/analytics/metadata.py [hunk 2]
```diff
@@ -101,32 +102,58 @@ def _generate_session_id() -> str:
...
 
-    # this can potentially be useful when generated on the host using the CLI and then mounted into the container via
-    # machine.json
+    # this can potentially be useful when generated on the host using the CLI and then mounted into the
+    # container via machine.json
     try:
         if os.path.exists("/etc/machine-id"):
             with open("/etc/machine-id") as fd:
-                return md5(str(fd.read()))[:8]
+                return f"sys_{md5(str(fd.read()))[:12]}"
```
Comment (ackdav) [inline localstack/utils/analytics/metadata.py:123]: nit: the ids will have different lengths - `short_uid()` has length 8, here we use 12. slightly neater if we unify them :shrug:

**Review by ackdav — APPROVED** [review]
Looks good to me - I'm not super deep into the crypto space, but I wanted to highlight that the [man pages of ubuntu](https://manpages.ubuntu.com/manpages/focal/man5/machine-id.5.html) suggest:
> If a stable unique identifier that is tied to the machine is needed for some application, the machine ID or any part of it must not be used directly. Instead the machine ID should be hashed with a cryptographic, keyed hash function, using a fixed, *application-specific key*.

if I understand it correctly, we're not using an application-specific key within `md5` yet. correct? what's your take?

File: localstack/utils/analytics/metadata.py (hunk 2 above)
Comment (thrau) [reply localstack/utils/analytics/metadata.py:123]: good point. i also made sure the prefixes are 3 characters long

File: localstack/utils/analytics/metadata.py (hunk 1 above)
Comment (thrau) [reply localstack/utils/analytics/metadata.py:109]: good idea, will add

**Pull request merged**